modules/service/src/main/resources/db/migration/V1036__blind_offset_asterism.sql

```diff
@@ -0,0 +1,162 @@
+-- add blind offset flag and target
+ALTER TABLE t_observation ADD COLUMN IF NOT EXISTS c_use_blind_offset boolean NOT NULL DEFAULT false;
+
+-- Create a function to enforce blind offset uniqueness
+CREATE OR REPLACE FUNCTION check_blind_offset_unique()
+RETURNS TRIGGER AS $$
+BEGIN
+  IF EXISTS (
+    SELECT 1 FROM t_target t
+    WHERE t.c_target_id = NEW.c_target_id
+    AND t.c_target_disposition = 'blind_offset'
+  ) THEN
+    -- Ensure no other blind offset target exists for this observation
+    IF EXISTS (
+      SELECT 1 FROM t_asterism_target a
+      JOIN t_target t ON a.c_target_id = t.c_target_id
+      WHERE a.c_observation_id = NEW.c_observation_id
+      AND t.c_target_disposition = 'blind_offset'
+      AND a.c_target_id != NEW.c_target_id
+    ) THEN
+      RAISE EXCEPTION 'Only one blind offset target allowed per observation';
+    END IF;
+  END IF;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Create trigger to enforce the constraint
+CREATE TRIGGER trg_blind_offset_unique
+  BEFORE INSERT OR UPDATE ON t_asterism_target
+  FOR EACH ROW
+  EXECUTE FUNCTION check_blind_offset_unique();
```

Comment:
Maybe this could be a `CONSTRAINT TRIGGER` that is `DEFERRABLE`. Seems like at some point we could need to make a number of changes in a transaction that will be eventually but not immediately correct.